src/apis/sykeApi.ts

```diff
@@ -1,35 +1,40 @@
 import { useMainStateStore } from '@/stores/mainStateStore'
 
-export default async function getVeslaData(resource: string, query: string) {
+const apiBase = 'https://rajapinnat.ymparisto.fi/api/meritietoportaali/api/'
+
+/** provide a generator to loop through response OData pages */
+export default async function* getPagedODataResponse(
```

Comment:
Originally I had the original getVeslaData side by side with this new paginated style, but for the life of me I couldn't figure out how to do the mocking with sinon. It only seemed to work when mocking the default export, otherwise it ignored the mock completely and sent the actual API requests.

However, maybe this is the more relevant default anyhow, and the changes in the usage side are rather minimal, so it's maybe not too bad